core/modules/node/node.block.inc

```diff
@@ -0,0 +1,163 @@
+   */
+  function getAdminPreview() {
+    return t('Displays content of node ID: @nid', array('@nid' => $this->settings['nid']));
+  }
+
+  /**
+   *  Sets block subject on block view.
+   */
+  function getTitle() {
+    $title = NULL;
+    $node = node_load($this->settings['nid']);
+    if ($this->settings['title_display'] === LAYOUT_TITLE_CUSTOM && $this->settings['title']) {
+      $title = check_plain($this->settings['title']);
+    }
+    elseif ($this->settings['title_display'] === LAYOUT_TITLE_DEFAULT) {
+      $title = $node->title;
+    }
+
+    if ($title && $this->settings['link_node_title']) {
+      $title = l($title, 'node/' . $node->nid);
+    }
+
+    return $title;
+  }
+
+  /**
+   *  Sets block content on block view.
+   */
+  function getContent() {
+    $node = node_load($this->settings['nid']);
+
+    // Prevent display of no access to the node.
+    if (!node_access('view', $node)) {
+      return;
+    }
+
+    // Use a clone of the node so we can hide the title safely.
+    $clone = clone($node);
+
+    // Hide node title if not required.
+    if (empty($this->settings['leave_node_title'])) {
+      $clone->title = '';
+    }
+
+    $content = node_view($clone, $this->settings['view_mode']);
+
+    // Hide links if they've been suppressed.
+    if (empty($this->settings['links'])) {
+      $content['links']['#access'] = FALSE;
+    }
+
+    // Add 'submitted by' settings.
+    if (!empty($this->settings['display_submitted'])) {
+      $content['#display_submitted'] = TRUE;
+    }
+
+    // Add theme hook suggestion.
+    $content['#theme'] = 'node__content-block-' . $this->settings['nid'];
+
+    return $content;
+  }
+
+  /**
+   *  Builds the block's settings configuration form.
+   */
+  function form(&$form, &$form_state) {
+    parent::form($form, $form_state);
+    $form['title_display']['title_display']['#options'][LAYOUT_TITLE_DEFAULT] = t('Use node title');
+    $form['title_display']['title_display']['#weight'] = -50;
+    $settings = $this->settings;
+
+    $form['nid'] = array(
+      '#title' => t('Enter the title or NID of a node'),
+      '#type' => 'textfield',
+      '#maxlength' => 512,
+      '#default_value' => !empty($settings['nid']) ? $settings['nid'] : NULL,
+      '#autocomplete_path' => 'node/autocomplete',
+    );
+
+    $form['display_submitted'] = array(
+      '#type' => 'checkbox',
+      '#default_value' => !empty($settings['display_submitted']),
+      '#title' => t('Show author and date information'),
+      '#description' => t('Advanced: if checked, display the node author and date information.'),
+    );
+
+    $form['leave_node_title'] = array(
+      '#type' => 'checkbox',
+      '#default_value' => !empty($settings['leave_node_title']),
+      '#title' => t('Show node title in block content'),
+      '#description' => t('Advanced: if checked, display the node title; this can cause the node title to appear twice unless you select "none" for the block title type.'),
```

Comment:
Again let's nix "Advanced" and remove redundant help text. Shorten to just
```
This may cause the node title to appear twice. Select "none" for the block title type if using this option.
```